applications/MeshingApplication/custom_processes/metrics_spr_error_process.cpp

```diff
@@ -0,0 +1,593 @@
+    --3-- Calculate metric (for each node) --3--
+    ******************************************************************************/
+
+    #pragma omp parallel for
+    for(int i_node = 0; i_node < num_nodes; ++i_node) {
+        auto it_node = nodes_array.begin() + i_node;
+        /**************************************************************************
+        ** Determine nodal element size h:
+        ** if mAverageNodalH == true : the nodal element size is averaged from the element size of neighboring elements
+        ** if mAverageNodalH == false: the nodal element size is the minimum element size from neighboring elements
+        */
+        double h_min = 0.0;
+        auto& neigh_elements = it_node->GetValue(NEIGHBOUR_ELEMENTS);
+        for(WeakElementItType i_neighbour_elements = neigh_elements.begin(); i_neighbour_elements != neigh_elements.end(); i_neighbour_elements++){
+            const double element_h = i_neighbour_elements->GetValue(ELEMENT_H);
+            if(mAverageNodalH == false) {
+                if(h_min == 0.0 || h_min > element_h)
+                    h_min = element_h;
+            } else {
+                h_min += element_h;
+            }
+        }
+
+        if(mAverageNodalH == true)
+            h_min = h_min/static_cast<double>(neigh_elements.size());
+
+        // Set metric
+        BoundedMatrix<double, TDim, TDim> metric_matrix = ZeroMatrix(TDim, TDim);
+        for(IndexType i = 0;i < TDim; ++i)
+            metric_matrix(i,i) = 1.0/std::pow(h_min, 2);
+
+        // Transform metric matrix to a vector
+        const Vector metric = MetricsMathUtils<TDim>::TensorToVector(metric_matrix);
+        it_node->SetValue(MMG_METRIC, metric);
+
+        KRATOS_INFO_IF("SPRMetricProcess", mEchoLevel > 2) << "Node "<<it_node->Id()<<" has metric: "<<it_node->GetValue(MMG_METRIC)<<std::endl;
+    }
+    
+    const double denominator = std::sqrt(std::pow(error_overall, 2) + std::pow(energy_norm_overall , 2));
+    const double coeff = denominator < tolerance ? 1.0 : 1.0/denominator;
+    KRATOS_WARNING_IF("SPRMetricProcess", denominator < tolerance) << "Denominator of error estimate zero or almost zero " << denominator << std::endl;
+    mThisModelPart.GetProcessInfo()[ERROR_ESTIMATE] = coeff * error_overall;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<SizeType TDim>
+void SPRMetricProcess<TDim>::CalculatePatch(
+    NodeItType itNode,
+    NodeItType itPatchNode,
+    SizeType NeighbourSize,
+    Vector& rSigmaRecovered
+    )
+{
+    // Determine if contact BC has to be regarded
+    // We take the geometry GP from the core
+    const double tolerance = std::numeric_limits<double>::epsilon();
+    const bool regard_contact = std::abs(itNode->GetValue(CONTACT_PRESSURE)) > tolerance ? true : false;
+    
+//     regard_contact = itPatchNode->Has(CONTACT_PRESSURE);
```

Comment:
leftover?

Reply:
Could be useful...

Reply:
Then please add a comment
I don't think it is good practice to add commented code, it makes it harder for others trying to understand your code